Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/consent/SubmissionConsentFragment.kt

```diff
@@ -46,11 +60,75 @@ class SubmissionConsentFragment : Fragment(R.layout.fragment_submission_consent)
                         requireActivity(),
                         REQUEST_USER_RESOLUTION
                     )
+                is SubmissionNavigationEvents.NavigateToDeletionWarningFragmentFromQrCode -> {
+                    doNavigate(
+                        NavGraphDirections
+                            .actionToSubmissionDeletionWarningFragment(
+                                it.consentGiven,
+                                it.coronaTestQRCode
+                            )
+                    )
+                }
             }
         }
         viewModel.countries.observe2(this) {
             binding.countries = it
         }
+
+        viewModel.showRedeemedTokenWarning.observe2(this) {
```

Comment:
btw you can use `observe` method now, with the latest version of the lifecycle library we will get a lint error if we tired to pass `this`. In Dialog fragments it is Ok to use `this` though . just mentioning it ,you don't have to change it 
cc/ @d4rken